bin/check-jenkins-health.rb

```diff
@@ -75,19 +75,42 @@ class JenkinsMetricsHealthChecker < Sensu::Plugin::Check::CLI
          proc: proc(&:to_i),
          default: 5
 
+  option :verbose,
+         short: '-v',
+         long: '--verbose',
+         boolean: true,
+         description: 'Return more verbose errors',
+         default: false
+
   def run
     https ||= config[:https] ? 'https' : 'http'
     testurl = "#{https}://#{config[:server]}:#{config[:port]}#{config[:uri]}"
 
-    r = if config[:https] && config[:insecure]
-          RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: false).get
-        elsif config[:https]
-          RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: true).get
-        else
-          RestClient::Resource.new(testurl, timeout: config[:timeout]).get
-        end
-
-    if [200, 500].include? r.code
+    begin
+      r = if config[:https] && config[:insecure]
+            RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: false)
+          elsif config[:https]
+            RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: true)
+          else
+            RestClient::Resource.new(testurl, timeout: config[:timeout])
+          end
+      if config[:verbose]
```

Comment:
cool, but the code doesn't work :(
I put the fixed code. Why didn't it take?
explanation:

```
r - it's instance of  RestClient::Resource
you are call  r.get and don't store the result of the query.
 
[200, 500].include?(r.code) this code doesn't valid
my code block

```
rest_client = if config[:https] && config[:insecure]
                      RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: false)
                    elsif config[:https]
                      RestClient::Resource.new(testurl, timeout: config[:timeout], verify_ssl: true)
                    else
                      RestClient::Resource.new(testurl, timeout: config[:timeout])
                    end
      r = if config[:verbose]
            rest_client.get
          else
            rest_client.get { |response| response }
          end
    rescue RestClient::SSLCertificateNotVerified => e
      critical "ssl verification failed: #{e.response}"
    rescue RestClient::ServerBrokeConnection
      critical "server broke the connection: #{e}"
    rescue RestClient::RequestFailed => e
      critical "request failed: #{e.response}"
    rescue RestClient::ExceptionWithResponse => e
      critical "failed to #{e.response}"
    end

    if [200, 500].include?(r.code)
```

Reply:
Sorry rather than copy/paste the entire code I just grabbed what I eyeballed. I will fix that up

Reply:
@yuri-zubov can you test #34 if I am still missing something just open a PR or attach a .patch to the PR and I will apply it. Sorry for the issues.

Reply:
Thanks for working through this you two! Note @majormoses as a maintainer you can choose `Edit` and get the raw next time. 

<img width="223" alt="screen shot 2018-03-01 at 9 38 26 pm" src="https://user-images.githubusercontent.com/1744971/36884958-eb6ac440-1d98-11e8-8fee-c237541f8ea8.png">

Reply:
ya my bad, could have handled that a bit better. I'd love it if someone would write an integration test so we could catch that kind of issue like that in the future.